pkg/stacks/unpack_test.go

```diff
@@ -969,122 +957,174 @@ status:
 
 ---
 `
-)
 
-var (
-	// Assert on test that *StackPackage implements StackPackager
-	_ StackPackager = &StackPackage{}
-)
-
-func simpleAppFile(permissionScope, packageType string, includeVersion bool) string {
-	appFile := `# apiVersion this app.yaml conforms to
-apiVersion: 0.1.0
-
-# Human readable title of application.
-title: Sample Crossplane Stack
-
-# Markdown description of this entry
-readme: |
- Markdown describing this sample Crossplane stack project.
-
-overview: text overview
-overviewShort: short text overview
-
-# Maintainer names and emails.
-maintainers:
-- name: Jared Watts
-  email: jared@upbound.io
-
-# Owner names and emails.
-owners:
-- name: Bassam Tabbara
-  email: bassam@upbound.io
-
-# Human readable company name.
-company: Upbound
-
-# Category name.
-category: Category
-
-dependsOn:
-- crd: "foo.mystack.example.org/v1alpha1"
-- crd: '*.yourstack.example.org/v1alpha2'
-
-# Keywords that describe this application and help search indexing
-keywords:
-- "samples"
-- "examples"
-- "tutorials"
-
-# Links to more information about the application (about page, source code, etc.)
-website: "https://upbound.io"
-source: "https://github.com/crossplane/sample-stack"
-packageType: %q
-permissionScope: %q
-
-# License SPDX name: https://spdx.org/licenses/
-license: Apache-2.0
-`
-
-	if includeVersion {
-		appFile += `
-# Version of project (optional)
-# If omitted the version will be filled with the docker tag
-# If set it must match the docker tag
-version: 0.0.1
-    `
+	if controllerImage != "" {
+		// The spaces are used for formatting the next line. This is a quick and dirty way
```

Comment:
If you replace:
```yaml
              %sname: sample-stack-controller
```
with 
```yaml
              %s#
              name: sample-stack-controller
```

Then this could become:
```go
imageTmpl := ""
if controllerImage != "" {
  imageTmpl = fmt.Sprintf("image: %s\n", controllerImage)
}
return fmt.Sprintf(tmpl, imageTmpl)
```

which would render the following (yaml doesn't care about the indentation of comment lines, right?):

```yaml
              image: foo/bar
#
              name: sample-stack-controller
```

Reply:
Yeah, that's true. I think that would work too. I decided to do it the way I've done it because I felt like I preferred the approach that kept the output looking nice, even if the formatting code looks a little weird.